browser/scripts/worldEditor/worldEditor.js

```diff
@@ -147,6 +167,97 @@ WorldEditor.prototype.setSelection = function(selected) {
 	}
 }
 
+WorldEditor.prototype.onPaste = function(nodes) {
+	if (!nodes || nodes.length < 1) {
+		return
+	}
+
+	var dropNode = nodes[0]
+
+	// find scene node
+	var sceneNode = this.scene.backReference.parentNode
+
+	// add a slot in the scene
+	E2.app.graphApi.addSlot(E2.core.root_graph, sceneNode, {
+		type: E2.slot_type.input,
+		name: sceneNode.getDynamicInputSlots().length + '',
+		dt: E2.dt.OBJECT3D
+	})
+
+	var slots = sceneNode.getDynamicInputSlots()
+	var slot = slots[slots.length - 1]
+
+	// connect the new patch to the scene
+	E2.app.graphApi.connect(E2.core.root_graph, Connection.hydrate(E2.core.root_graph, {
+		src_nuid: dropNode.uid,
+		dst_nuid: sceneNode.uid,
+		src_slot: 0,
+		src_dyn: true, // TODO: the src slot is not necessarily a dyn slot
+		dst_slot: slot.index,
+		dst_dyn: true
+	}))
+
+	// set a pending selection object for the pasted objects
+	// we have to wait for one update to pass before actually
+	// setting the selection, because mesh creation itself is
+	// deferred in AbstractThreeMeshPlugin
+	var pendingSelection = {waitTime: 1, selection: []}
+
+	function collectMeshes(node) {
+		if (node.plugin && node.plugin.object3d) {
+			pendingSelection.selection.push(node)
+		}
+
+		if (node.plugin.graph) {
+			for (var n = 0; n < node.plugin.graph.nodes.length; ++n) {
+				collectMeshes(node.plugin.graph.nodes[n])
+			}
+		}
+	}
+
+	for (var i = 0; i < nodes.length; ++i) {
+		var node = nodes[i]
+		collectMeshes(node)
+	}
+
+	this.pendingSelection = pendingSelection
+
+	// TODO: if this.pendingSelection.length === 0, we didn't paste any objects
+	// and we could warn the user somehow
+}
+
+WorldEditor.prototype.selectMeshAndDependencies = function(meshNode, sceneNode) {
+	function markNodeSelectedIfConnectedTo(curNode, endNode) {
+		var allOutputs = curNode.outputs.concat(curNode.dyn_outputs)
+
+		if (curNode.id === 'Output proxy' && curNode.parent_graph && curNode.parent_graph.plugin && curNode.parent_graph.plugin.parentNode) {
```

Comment:
where is this set as 'Output proxy'? should this compare `curNode.plugin.id` to 'output_proxy' instead?